몽고디비 접속 문제

ACL 은 telnet 으로 확인했는데 접속이 안됨.
에러 로그
[deploy@xmllog-aggregator-real ~]$ mongo --host voice_mongo_key-prod/10.182.29.185:27017,10.182.30.159:27017,10.182.30.199:27017 --username voice --password --authenticationDatabase voice_mongolog_keys voice_mongolog_keys
MongoDB shell version v3.4.13
Enter password:
connecting to: mongodb://10.182.29.185:27017,10.182.30.159:27017,10.182.30.199:27017/voice_mongolog_keys?replicaSet=voice_mongo_key-prod
2022-01-14T09:51:00.815+0000 I NETWORK [thread1] Starting new replica set monitor for voice_mongo_key-prod/10.182.29.185:27017,10.182.30.159:27017,10.182.30.199:27017
2022-01-14T09:51:00.817+0000 I NETWORK [ReplicaSetMonitor-TaskExecutor-0] Successfully connected to 10.182.30.159:27017 (1 connections now open to 10.182.30.159:27017 with a 5 second timeout)
2022-01-14T09:51:00.817+0000 I NETWORK [thread1] Successfully connected to 10.182.30.199:27017 (1 connections now open to 10.182.30.199:27017 with a 5 second timeout)
2022-01-14T09:51:00.843+0000 I NETWORK [thread1] Detected bad connection created at 1642153860816699 microSec, clearing pool for 10.182.30.199:27017 of 0 connections
2022-01-14T09:51:00.843+0000 I NETWORK [thread1] Dropping all pooled connections to 10.182.30.199:27017(with timeout of 5 seconds)
2022-01-14T09:51:00.843+0000 I NETWORK [thread1] Ending connection to host 10.182.30.199:27017(with timeout of 5 seconds) due to bad connection status; 0 connections to that host remain open
2022-01-14T09:51:00.843+0000 I NETWORK [ReplicaSetMonitor-TaskExecutor-0] Detected bad connection created at 1642153860816672 microSec, clearing pool for 10.182.30.159:27017 of 0 connections
2022-01-14T09:51:00.843+0000 I NETWORK [ReplicaSetMonitor-TaskExecutor-0] Dropping all pooled connections to 10.182.30.159:27017(with timeout of 5 seconds)
2022-01-14T09:51:00.843+0000 I NETWORK [ReplicaSetMonitor-TaskExecutor-0] Ending connection to host 10.182.30.159:27017(with timeout of 5 seconds) due to bad connection status; 0 connections to that host remain open

가 반복됨.

잘 살펴보면 서버 연결은 되었는데 접속이 안되는 문제

에러 MongoDB Source Server MongoDB version

[deploy@xmllog-aggregator-real ~]$ mongo --version
MongoDB shell version v3.4.13
git version: fbdef2ccc53e0fcc9afb570063633d992b2aae42
OpenSSL version: OpenSSL 1.0.1e-fips 11 Feb 2013
allocator: tcmalloc
modules: none
build environment:
distmod: rhel70
distarch: x86_64
target_arch: x86_64

업데이트 버전

[deploy@xmllog-aggregator-real mongodb-4.2]$ mongo --version
MongoDB shell version v5.0.5
Build Info: {
"version": "5.0.5",
"gitVersion": "d65fd89df3fc039b5c55933c0f71d647a54510ae",
"openSSLVersion": "OpenSSL 1.0.1e-fips 11 Feb 2013",
"modules": [],
"allocator": "tcmalloc",
"environment": {
"distmod": "rhel70",
"distarch": "x86_64",
"target_arch": "x86_64"
}
}

Destination MongoDB version

[deploy@usrlm-voice-mongo-prod-01 ~]$ mongo --version
MongoDB shell version v4.2.14
git version: 0e6db36e92d82cc81cbd40ffd607eae88dc1f09d
OpenSSL version: OpenSSL 1.0.1e-fips 11 Feb 2013
allocator: tcmalloc
modules: none
build environment:
distmod: rhel70
distarch: x86_64
target_arch: x86_64

참고 : https://www.mongodb.com/community/forums/t/unable-to-connect-to-cluster-no-primary-detected/90700/17

버전 차이 문제이다.
